tests/integration/dataframe/test_dataframe.py

```diff
@@ -615,3 +628,217 @@ def test__export_table_data__works(self, client: DataFrameClient, create_table):
                 response.read()
                 == b'"col1","col2","col3"\r\n1,2.5,6.5\r\n2,1.5,5.5\r\n3,2.5,7.5'
             )
+
+    def test__append_table_data__append_request_success(
+        self, client: DataFrameClient, create_table
+    ):
+        table_id = self._new_single_int_table(create_table)
+        frame = DataFrame(columns=["a"], data=[["1"], ["2"]])
+        client.append_table_data(
+            table_id, AppendTableDataRequest(frame=frame, end_of_data=True)
+        )
+
+    def test__append_table_data__append_request_with_end_of_data_argument_disallowed(
+        self, client: DataFrameClient, create_table
+    ):
+        request = AppendTableDataRequest(end_of_data=True)
+        with pytest.raises(
+            ValueError,
+            match="end_of_data must not be provided separately when passing an AppendTableDataRequest.",
+        ):
+            client.append_table_data(
+                self._new_single_int_table(create_table), request, end_of_data=True
+            )
+
+    def test__append_table_data__append_request_without_end_of_data_success(
+        self, client: DataFrameClient, create_table
+    ):
+        table_id = self._new_single_int_table(create_table)
+        frame = DataFrame(columns=["a"], data=[["7"], ["8"]])
+        client.append_table_data(table_id, AppendTableDataRequest(frame=frame))
+
+    def test__append_table_data__accepts_dataframe_model(
+        self, client: DataFrameClient, create_table
+    ):
+        table_id = self._new_single_int_table(create_table)
+        frame = DataFrame(columns=["a"], data=[["1"], ["2"]])
+        client.append_table_data(table_id, frame, end_of_data=True)
+
+    def test__append_table_data__dataframe_without_end_of_data_success(
+        self, client: DataFrameClient, create_table
+    ):
+        table_id = self._new_single_int_table(create_table)
+        frame = DataFrame(columns=["a"], data=[["10"], ["11"]])
+        client.append_table_data(table_id, frame)
+
+    def test__append_table_data__none_without_end_of_data_raises(
+        self, client: DataFrameClient, create_table
+    ):
+        table_id = create_table(basic_table_model)
+        with pytest.raises(
+            ValueError, match="end_of_data must be provided when data is None"
+        ):
+            client.append_table_data(table_id, None)
+
+    def test__append_table_data__flush_only_with_none(
+        self, client: DataFrameClient, create_table
+    ):
+        table_id = self._new_single_int_table(create_table)
+        frame = DataFrame(columns=["a"], data=[["1"]])
+        client.append_table_data(table_id, frame)
```

Comment:
nit: Appending data before setting end of data isn't required. The test could verify `RowCount` is 0 and `SupportsAppend` is false.